### PR TITLE
Team forming: all agents refuse to join a team #59

### PR DESCRIPTION
**Pull request by sodabreak**

### CODE REVIEW

File: server/EnvironmentServer.go [hunk 1]
```diff
@@ -343,9 +343,35 @@ func (cs *EnvironmentServer) StartAgentTeamForming() {
...
+			}
+		}
+		attempts++
+	}
+
+	// Server intervention: assign agents without teams to existing teams
+	if !allAgentsAttempted {
+		for _, agent := range cs.GetAgentMap() {
+			if agent.GetTeamID() == uuid.Nil {
+				for _, team := range cs.teams {
```
Comment (arnavxkohli) [inline server/EnvironmentServer.go:367]: Perhaps we want to allot them round-robin based on team sizes? So the smallest team is allotted the agent first? Not sure how much this matters, but if we are sticking to fixed size teams this should be implemented

**Review by arnavxkohli — CHANGES_REQUESTED** [review]

File: server/EnvironmentServer.go (hunk 1 above)
Comment (nlewxxs) [reply server/EnvironmentServer.go:367]: I'm assuming teams cannot reject agents at this point? At the start of the iteration are all agent 'opinions' cleared? 

I'm trying to see if there is a case here where agents get rejected by the team during team formation but the server allocates them to the team anyway. 

I might be misunderstanding how team formation works.

**Comment by nlewxxs** [comment]
I'm assuming teams cannot reject agents at this point? At the start of the iteration are all agent 'opinions' cleared?

I'm trying to see if there is a case here where agents get rejected by the team during team formation but the server allocates them to the team anyway.

I might be misunderstanding how team formation works.

**Review by kushk2 — APPROVED** [review]

**Comment by dhruvds12** [comment]
@sodabreak I think these changes were made in a later PR. 

Closing this PR

**Pull request closed without merging**